benchmark/bench_indexed_persistence.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+# SPDX-FileCopyrightText: Copyright (c) 2024-2025 Yegor Bugayenko
+# SPDX-License-Identifier: MIT
+
+require_relative '../lib/factbase'
+
+def bench_indexed_persistence(bmk, _fb)
+  total = 5_000
+  fb_source = Factbase::IndexedFactbase.new(Factbase.new)
```

Comment:
@HaidarJbeily7 as far as I know, we have already create an `IndexedFactbase` for each benchmark. Maybe we can use `_fb` param instead of `fb_source`? 
If you need to be sure that we are using `IndexedFactbase`, you can use `fb.is_a?(IndexedFactbase)`.